### ISSUE TITLE
dotnet build is slower on Linux than Windows

### ISSUE THREAD

**Issue by mikeharder** [dotnet/cli#3146]
## Steps to reproduce
1. Create Linux VM in Hyper-V
2. Run the following commands on both Windows and Linux
 1. `dotnet new`
 2. `dotnet restore`
 3. `set DOTNET_CLI_CAPTURE_TIMING=1` (Windows) or `export DOTNET_CLI_CAPTURE_TIMING=1` (Linux)
 4. `dotnet build`

## Expected  behavior
Build time is about the same on Windows and Linux.

### Windows
```
Project new (.NETCoreApp,Version=v1.0) will be compiled because expected outputs are missing
Compiling new for .NETCoreApp,Version=v1.0

Compilation succeeded.
    0 Warning(s)
    0 Error(s)

Time elapsed 00:00:01.5097798

Performance Summary:
[100.00% 01.906s] Thread 1
  [99.82% 99.82% 01.903s] Program:Main
    [07.74% 07.76% 00.147s] Program:ResolveRootContexts
      [03.49% 45.07% 00.066s] Program:Loading project.json C:\Temp\new\project.json
      [03.96% 51.16% 00.075s] Program:Waiting for project contexts to finish loading
    [00.07% 00.07% 00.001s] Program:Collect graph
    [90.62% 90.78% 01.728s] ProjectBuilder:Build new
      [00.80% 00.88% 00.015s] ProjectBuilder:HasSourceFiles new
      [02.65% 02.93% 00.050s] ProjectBuilder:NeedsRebuilding new
      [87.02% 96.02% 01.659s] ProjectBuilder:RunCompile new
        [74.06% 85.11% 01.412s] Command:Execute dotnet "C:\Program Files\dotnet\sdk\1.0.0-preview1-002702\csc.dll" -noconfig @C:\Temp\new\obj\Debug\netcoreapp1.0\dotnet-compile-csc.rsp
```

### Linux
```
Project new (.NETCoreApp,Version=v1.0) will be compiled because expected outputs are missing
Compiling new for .NETCoreApp,Version=v1.0

Compilation succeeded.
    0 Warning(s)
    0 Error(s)

Time elapsed 00:00:03.5143866

Performance Summary:
[100.00% 04.528s] Thread 1
  [99.98% 99.98% 04.527s] Program:Main
    [10.03% 10.03% 00.453s] Program:ResolveRootContexts
      [05.34% 53.26% 00.241s] Program:Loading project.json /tmp/new/project.json
      [04.31% 42.95% 00.194s] Program:Waiting for project contexts to finish loading
    [00.20% 00.20% 00.009s] Program:Collect graph
    [88.57% 88.59% 04.011s] ProjectBuilder:Build new
      [00.35% 00.39% 00.015s] ProjectBuilder:HasSourceFiles new
      [02.33% 02.63% 00.105s] ProjectBuilder:NeedsRebuilding new
      [85.81% 96.88% 03.885s] ProjectBuilder:RunCompile new
        [73.26% 85.38% 03.317s] Command:Execute dotnet /usr/share/dotnet/sdk/1.0.0-preview1-002702/csc.dll -noconfig @/tmp/new/obj/Debug/netcoreapp1.0/dotnet-compile-csc.rsp
```
## Environment data
`dotnet --info` output:

### Windows
```
.NET Command Line Tools (1.0.0-preview1-002702)

Product Information:
 Version:     1.0.0-preview1-002702
 Commit Sha:  6cde21225e

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.10586
 OS Platform: Windows
 RID:         win10-x64
```

### Linux
```
.NET Command Line Tools (1.0.0-preview1-002702)

Product Information:
 Version:     1.0.0-preview1-002702
 Commit Sha:  6cde21225e

Runtime Environment:
 OS Name:     debian
 OS Version:  8
 OS Platform: Linux
 RID:         debian.8-x64
```

**Comment by piotrpMSFT** [comment]
@brianrob @lt72 are we collecting comparable data across RIDs?

**Comment by cdmihai** [comment]
It looks like all the execution is slower. Is the pre-jitted CLI code being picked up on Linux?

**Comment by brianrob** [comment]
My understanding is that there were some recent issues that caused crossgen to be disabled.  I believe that it is being re-enabled at this time.  @eerhardt has details.

**Comment by eerhardt** [comment]
Crossgen was only disabled on Windows in the latest CLI builds, not on Linux.

We are working on getting a new CLI build out that has crossgen enabled on Windows. The code is in, just need the official build.

**Comment by brianrob** [comment]
That's a bit worrisome then.  @eerhardt can you let us know when there is a version of CLI that has crossgen enabled and Ican confirm that it's doing the right thing on Linux as well?

**Comment by eerhardt** [comment]
@brianrob - the latest CLI version: 

b3bf977c008666cb435e43377255c3ed54843def
1.0.0-preview2-002911

Has all the latest updates with crossgen fully enabled across all platforms.  Please try it out, and let me know if you hit any issues.

**Comment by brianrob** [comment]
Thanks @eerhardt.  It does appear that there is a ton of JIT compilation that I would not have expected to be present consider that ready-to-run is being used by the CLI and shared framework.

@gkhanna79, @JohnChen0, @fadimounir, can you please take a look at this?

**Comment by brianrob** [comment]
Assigning to @JohnChen0 for further investigation.

**Comment by fadimounir** [comment]
The fix for this bug is merged into dotnet/coreclr master branch (922cace885bfe24c18508747346e23ff7dc40d6c), and ported to the dotnet/coreclr release/1.0.0 branch (cc2e4db2c66bfac567f222cd302edaccc4c5d3cf). Closing issue.
